### PR TITLE
Create Migration notes for for 6.x to 7.x

### PR DESCRIPTION
**Pull request by paxnovem**
Provides migrating versions #1558

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -52,3 +52,4 @@ tmp
 .DS_Store
 
 !.github
+.devcontainer/devcontainer.json
```
Comment (nedbat) [inline .gitignore]: Do we need this file? It sounds like something that should be in a global git ignore. I'm not familiar with it.

File: doc/migrations.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
+.. Licensed under the Apache License: http://www.apache.org/licenses/LICENSE-2.0
+.. For details: https://github.com/nedbat/coveragepy/blob/master/NOTICE.txt
+
+.. Licensed under the Apache License: http://www.apache.org/licenses/LICENSE-2.0
+.. For details: https://github.com/nedbat/coveragepy/blob/master/NOTICE.txt
+
```
Comment (nedbat) [inline doc/migrations.rst]: This is doubled.

File: doc/migrations.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+.. _migrations:
+
+==============================
+Migrating between versions
+==============================
+
+.. _migrating_6x_7x:
+
+Migrating 6.5.x — 7.0.x
+--------------------------
```
Comment (nedbat) [inline doc/migrations.rst]: Add a blank line, and please make the rules match the length of the titles.

File: doc/migrations.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+- The way that wildcards when specifing file paths work in certain cases as changed in 7.0.x:
```
Comment (nedbat) [inline doc/migrations.rst]:
```suggestion
- The way that wildcards when specifing file paths work in certain cases has changed in 7.0.x:
```

**Review by nedbat — CHANGES_REQUESTED** [review]

File: doc/migrations.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  - Previously, ``*`` would incorrectly match directory separators, making
+    precise matching difficult. This requires patterns such as ``*tests/*``
+    to be changed to ``*/tests/*``.
```
Comment (nedbat) [inline doc/migrations.rst]:
```suggestion
    precise matching difficult. Patterns such as ``*tests/*``
    will need to be changed to ``*/tests/*``.
```

File: .gitignore (hunk 1 above)
Comment (paxnovem) [reply .gitignore]: I added that to prevent my local environment from getting committed. I removed this and locally used `.git/info/exclude` instead.

**Comment by nedbat** [comment]
I resolved the conflicts and merged it in commit d6e621aa. Thanks!

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.2.4](https://pypi.org/project/coverage/7.2.4).

**Pull request closed without merging**